index.js

```diff
@@ -20,7 +20,11 @@ function decirc (val, k, stack, parent) {
   if (typeof val === 'object' && val !== null) {
     for (i = 0; i < stack.length; i++) {
       if (stack[i] === val) {
-        parent[k] = '[Circular]'
+        if (Object.getOwnPropertyDescriptor(parent, k).get !== undefined) {
```

Comment:
We have to make sure to set the reset the property descriptor after calling `JSON.stringify`. Otherwise the input object would not be identical to the original object. That is a bug in the current implementation, since it did not pay attention to the property descriptor.

It seems like the current implementation will also always fail in case `configurable` is set to `false`.

Reply:
Right, I'll look into after work tonight!